Precompiled lib, use webpack and update dependency versions

scags9876 opened the pull request:
Some useful stuff in here, thought others might benefit from these changes.

 - using webpack instead of grunt and browserify
 - including a prepublish script to precompile transpiled code and include it in a lib directory in the package published on npm
 - upgrading dependencies to work with react 15
 - upgrade to lodash 4.x

he8us committed:
Added Prepublish hook

Moved from browserify and grunt to webpack
Use babel with react preset
Npm based build

jbrodriguez committed:
Publish precompiled lib

scags9876 committed:
Upgrade to work with react 15

- Set react dependencies to >=0.14.x so that it will accept react 15

scags9876 committed:
Upgrade lodash to 4.x

scags9876 committed:
Bump version to 1.5.2

scags9876 committed:
Add some docs for the development scripts

scags9876 committed:
update bundle

scags9876 committed:
Remove compiled lib from git

scags9876 commented:
Addresses:

#23
#19
#21

jbrodriguez commented:
@scags9876 have you published your 'fork' to npm ?

If that's not the case, how can one use your code directly from github (since the compiled lib doesn't exist in github)

scags9876 commented:
@jbrodriguez  I have not published to npm... I'm waiting for @MandarinConLaBarba to merge this PR so he can publish.

In the meantime, i've got the following in my package.json:

```
"react-tree-menu": "git://github.com/scags9876/react-tree-menu.git",
```

It points to my master branch on my fork, which has the precompiled lib... see https://github.com/MandarinConLaBarba/react-tree-menu/pull/31#issuecomment-280138789

scags9876 committed:
Update to support react v15

These mainly address deprecation warnings given with react 15.

 - Remove soon to be deprecated React.propTypes in favor of the PropTypes package.  See https://fb.me/prop-types-docs
 - Remove soon to be deprecated React.createClass in favor of the creatReactClass package. See https://fb.me/react-create-class
 - Fix bug with undefined propType warning

scags9876 commented:
this PR now includes the fix for the undefined PropType from https://github.com/MandarinConLaBarba/react-tree-menu/pull/33

xitarps commented:
> @jbrodriguez I have not published to npm... I'm waiting for @MandarinConLaBarba to merge this PR so he can publish.
> 
> In the meantime, i've got the following in my package.json:
> 
> ```
> "react-tree-menu": "git://github.com/scags9876/react-tree-menu.git",
> ```
> 
> It points to my master branch on my fork, which has the precompiled lib... see [#31 (comment)](https://github.com/MandarinConLaBarba/react-tree-menu/pull/31#issuecomment-280138789)

Thanks!! helped me a lot!! /o/
Mandou muito! (BR)
